README.md

```diff
@@ -87,20 +105,39 @@ This step command will generate an slsa attestation provenance ``xygeni-salt-att
 
-**Pipeline Syntax tool** could helps to define ``xygeniSaltSlsa`` step with their arguments. 
+**Pipeline Syntax tool** could helps to define ``xygeniSalt`` steps with their arguments. 
 
-The ``xygeniSaltSlsa`` step could be invoked for generating SALT provenance. Build information for the registered attestation **subjects** (also known as software 'products' or 'artifacts') will be registered in the signed attestation.
+ * The ``xygeniSaltSlsa`` step could be invoked for generating SALT provenance. Build information for the registered attestation **subjects** (also known as software 'products' or 'artifacts') will be registered in the signed attestation.
+
+### Step parameters
+
+#### Subjects 
 
 Subjects could be provided explicitly as a list in the ``subjects`` property. Each item in the list is a map with a given ``name`` and a content, either a Docker image published in a remote registry as part of the build (``image: 'REGISTRY/IMAGE_NAME:TAG'``), a local file produced by the build as a packaged artifact (``file: 'path/to/file.zip'``), or a value (which could be a SHA digest of a given artifact, a base-64 encoded binary value, or a string representing the artifact (``value: 'sha:03afb3...1c24'``).
 
 Alternatively, subjects could be referenced by pattern using the ``artifactFilter`` pattern (an Ant-like pattern), which matches files in the workspace that will be used as subjects for the SLSA provenance attestation.
 
+#### Signer configuration
+
 The ``key`` / ``publicKey`` parameters contain the key pair to use for signing the provenance file. The signature is done with the private key, and the public key is added to the signed attestation for verification by software consumers. The keys are provided as either PEM-encoded values (text format with key enclosed between ``-----BEGIN...-----`` and ``-----END ...-----`` delimiters). The PEM-encoded key could be provided also as a path to the key file relative to the workspace directory (prefix the path with ``file:`` prefix), or as an environment variable prefixed with ``env:``.
 
 An optional X.509 certificate could be used for helping the software consumer to trust the signature, using the ``certificate`` parameter with a similar format.
 
 The ``pkiFormat`` specifies the signature format (one of ``x509``, ``minisign``, ``ssh``, ``pkcs7`` or ``tuf``). Use ``x509`` as a good default.
 
+Remember to encrypt the signing private ``key`` with a strong password !
+For the ``key-password``, create a Jenkins secret and use the secret name as value for the field.
+
+You can use ``xygeniSalt keygen`` command option to generate and save signing keys to use in this attestation command.
+
+![xygenisalt-signer.png](docs/images/xygenisalt-signer.png)
+
+#### Keyless signatures
+
+To use keyless signing, the `--keyless` option could be passed to the attestation commit | provenance commands. This will fetch an OpenID Connect (OIDC) from specific issuers, using an OAuth flow (standard or device).
```

Comment:
More specific info about the 'keyless' option should be added here, in the context of the Jenkins pipeline where salt commands runs (where the interactive authentication flow is not available). Salt commit / provenance commands allow a --keyless option . GitHub and GitLab provide their own OIDC providers. See https://docs.xygeni.io/xydocs/build_security/concepts.html#_keyless_signing and https://docs.xygeni.io/xydocs/build_security/salt_how_to.html#_use_keyless_signatures for more details.

I have not tested keyless with Jenkins (needs the OIDC plugin configured appropriately), but it could be a nice thing to do.

My suggestion follows:

To use keyless signing, the `--keyless` option could be passed to the salt `commit` | `provenance` commands. An ephemeral keypair is generated, and an ephemeral X.509 certificate will be issued by the [SigStore Fulcio CA](https://docs.sigstore.dev/certificate_authority/overview/). 

To authenticate the Jenkins build, Fulcio needs an OpenID Connect (OIDC) id token, as explained in [OIDC Usage in Fulcio](https://docs.sigstore.dev/certificate_authority/oidc-in-fulcio/). Such OIDC token could be generated using the Jenkins [OpenID Connect Provider plugin](https://plugins.jenkins.io/oidc-provider/). The claims to be included in the token for proper authentication with Fulcio CA are `iss`, the issuer (which should be set to `https://oauth2.sigstore.dev/auth`) and `aud`, the audience (which should be set to `sigstore`). The `sub`, subject  is set by the plugin as the default URL of the Jenkins job, which is adequate as the "identity" of the signer in this context. See [Registering the identity provider](https://oauth2.sigstore.dev/auth) for further information.

Jobs running in Jenkins runners use the plugin's OIDC provider that authenticates the job and generates a short-lived OIDC ID token. The OIDC plugin will store the token either in an environment variable or in a CI/CD file, so the `xygeniSalt commit` or `xygeniSaltSlsa` commands will fetch it to use for authentication with Fulcio CA.The variable and the path are configured in the `conf/salt.yaml` file, and the default values are `SIGSTORE_ID_TOKEN` or `/var/run/sigstore/cosign/oidc-token` path, respectively.

The ephemeral certificate has a life of 10 minutes, and the event is registered in a transparency log. Verifiers of the attestation can check that the certificate was valid at the moment where the attestation was signed, and validate the certificate chain. The identity (the Jenkins build URI) will be registered as an URI field in the X509 certificate's SAN (Subject Alternative Name), so the verifier knows which jenkins job created the attestation. For full details, read [Certificate Issuing Overview](https://docs.sigstore.dev/certificate_authority/certificate-issuing-overview/).